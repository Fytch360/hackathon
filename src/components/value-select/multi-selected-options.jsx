import React, { useCallback } from 'react'
import PropTypes from 'prop-types'

import { ic24CrossSmall } from '../icon/common'
import { disableHandler } from '../utils/handlers'
import { lineHeightLg, lineHeightMd } from '../styles/font-sizes.config.style'

import {
    MultiSelectedListStyled,
    MultiSelectedItemStyled,
    MultiSelectedTitleStyled,
    MultiSelectedButtonStyled,
    MultiSelectedIconStyled
} from './value-select.style'

export const MultiSelectedOptions = ({ options, onChange, size, disabled }) => {
    const handleClick = useCallback(
        (value) => (event) => {
            event.stopPropagation()

            const newSelected = options
                .map((option) => option.value)
                .filter((val) => val !== value)
            onChange(newSelected)
        },
        [options]
    )

    return (
        <MultiSelectedListStyled>
            {options.map((option) => (
                <MultiSelectedItemStyled key={option.value} size={size} disabled={disabled}>
                    <MultiSelectedTitleStyled size={size} fontWeight="medium" disabled={disabled}>
                        {option.title}
                    </MultiSelectedTitleStyled>

                    {!disabled && <MultiSelectedButtonStyled
                        role="button"
                        aria-label="Закрыть"
                        tabIndex={0}
                        size={size}
                        onClick={disableHandler(handleClick(option.value), disabled)}
                    >
                        <MultiSelectedIconStyled
                            icon={ic24CrossSmall}
                            width={10}
                            height={parseInt(size === 'lg' ? lineHeightLg : lineHeightMd, 10)}
                        />
                    </MultiSelectedButtonStyled>}
                </MultiSelectedItemStyled>
            ))}
        </MultiSelectedListStyled>
    )
}

MultiSelectedOptions.propTypes = {
    options: PropTypes.arrayOf(PropTypes.object).isRequired,
    onChange: PropTypes.func.isRequired,
    size: PropTypes.string.isRequired,
    disabled: PropTypes.bool
}
