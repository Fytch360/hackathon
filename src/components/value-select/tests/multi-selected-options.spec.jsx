import React from 'react'
import { shallow } from 'enzyme'

import { MultiSelectedOptions } from '../multi-selected-options'

describe('MultiSelectedOptions', () => {
    const optionsExample = [
        { title: 'Title 0', value: 'Value 0' },
        { title: 'Title 1', value: 'Value 1' },
        { title: 'Title 2', value: 'Value 2' },
    ]
    const propsExample = {
        size: 'lg',
        options: optionsExample,
    }
    let onChangeMock
    let wrapper

    beforeEach(() => {
        onChangeMock = jest.fn()

        wrapper = shallow(<MultiSelectedOptions {...propsExample} onChange={onChangeMock} />)
    })

    test('Should be available', () => {
        expect(MultiSelectedOptions).toBeDefined()
    })

    test('Snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should\'nt render close button if gets disabled prop', () => {
        wrapper.setProps({ disabled: true })

        expect(wrapper.find('MultiSelectedButtonStyled').exists()).toBeFalsy()
    })

    test('Should\'nt call onChange w/o any actions', () => {
        expect(onChangeMock.mock.calls).toHaveLength(0)
    })

    test('Should call onChange if option remove button was clicked', () => {
        wrapper.children().last().find('MultiSelectedButtonStyled')
            .simulate('click', { stopPropagation: jest.fn() })

        expect(onChangeMock.mock.calls).toHaveLength(1)
    })

    test('Should return updated option values array if option remove button was clicked', () => {
        wrapper.childAt(1).find('MultiSelectedButtonStyled')
            .simulate('click', { stopPropagation: jest.fn() })

        expect(onChangeMock.mock.calls[0][0]).toEqual(
            expect.not.arrayContaining([optionsExample[1].value])
        )
    })
})
