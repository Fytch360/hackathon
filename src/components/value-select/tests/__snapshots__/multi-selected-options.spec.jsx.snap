// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectedOptions Snapshot 1`] = `
<MultiSelectedListStyled>
  <MultiSelectedItemStyled
    key="Value 0"
    size="lg"
  >
    <MultiSelectedTitleStyled
      fontWeight="medium"
      size="lg"
    >
      Title 0
    </MultiSelectedTitleStyled>
    <MultiSelectedButtonStyled
      aria-label="Закрыть"
      onClick={[Function]}
      role="button"
      size="lg"
      tabIndex={0}
    >
      <MultiSelectedIconStyled
        height={24}
        icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M15.293 7.293L12 10.585 8.707 7.293c-.36-.36-.928-.388-1.32-.083l-.094.083c-.39.39-.39 1.024 0 1.414L10.585 12l-3.292 3.293c-.39.39-.39 1.024 0 1.414.39.39 1.024.39 1.414 0L12 13.415l3.293 3.292c.36.36.928.388 1.32.083l.094-.083c.39-.39.39-1.024 0-1.414L13.415 12l3.292-3.293c.39-.39.39-1.024 0-1.414-.39-.39-1.024-.39-1.414 0z\\"/></svg>"
        width={10}
      />
    </MultiSelectedButtonStyled>
  </MultiSelectedItemStyled>
  <MultiSelectedItemStyled
    key="Value 1"
    size="lg"
  >
    <MultiSelectedTitleStyled
      fontWeight="medium"
      size="lg"
    >
      Title 1
    </MultiSelectedTitleStyled>
    <MultiSelectedButtonStyled
      aria-label="Закрыть"
      onClick={[Function]}
      role="button"
      size="lg"
      tabIndex={0}
    >
      <MultiSelectedIconStyled
        height={24}
        icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M15.293 7.293L12 10.585 8.707 7.293c-.36-.36-.928-.388-1.32-.083l-.094.083c-.39.39-.39 1.024 0 1.414L10.585 12l-3.292 3.293c-.39.39-.39 1.024 0 1.414.39.39 1.024.39 1.414 0L12 13.415l3.293 3.292c.36.36.928.388 1.32.083l.094-.083c.39-.39.39-1.024 0-1.414L13.415 12l3.292-3.293c.39-.39.39-1.024 0-1.414-.39-.39-1.024-.39-1.414 0z\\"/></svg>"
        width={10}
      />
    </MultiSelectedButtonStyled>
  </MultiSelectedItemStyled>
  <MultiSelectedItemStyled
    key="Value 2"
    size="lg"
  >
    <MultiSelectedTitleStyled
      fontWeight="medium"
      size="lg"
    >
      Title 2
    </MultiSelectedTitleStyled>
    <MultiSelectedButtonStyled
      aria-label="Закрыть"
      onClick={[Function]}
      role="button"
      size="lg"
      tabIndex={0}
    >
      <MultiSelectedIconStyled
        height={24}
        icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M15.293 7.293L12 10.585 8.707 7.293c-.36-.36-.928-.388-1.32-.083l-.094.083c-.39.39-.39 1.024 0 1.414L10.585 12l-3.292 3.293c-.39.39-.39 1.024 0 1.414.39.39 1.024.39 1.414 0L12 13.415l3.293 3.292c.36.36.928.388 1.32.083l.094-.083c.39-.39.39-1.024 0-1.414L13.415 12l3.292-3.293c.39-.39.39-1.024 0-1.414-.39-.39-1.024-.39-1.414 0z\\"/></svg>"
        width={10}
      />
    </MultiSelectedButtonStyled>
  </MultiSelectedItemStyled>
</MultiSelectedListStyled>
`;
